src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1409,6 +1410,25 @@ void PosixSignals::print_signal_handler(outputStream* st, int sig,
   print_single_signal_handler(st, sig, &current_act, buf, buflen);
   st->cr();
 
+  // If we expected to see our own hotspot signal handler but found a different one,
+  //  print a warning. Unless it had been replaced with our own secondary crash handler.
+  const struct sigaction* expected_act = expected_handlers.get(sig);
+  if (expected_act != NULL) {
+    const address current_handler = get_signal_handler(&current_act);
+    if (!(HANDLER_IS(current_handler, VMError::crash_handler_address))) {
+      const address expected_handler = get_signal_handler(expected_act);
+      if (current_handler != expected_handler) {
+        st->print_cr("  *** Handler changed! Expected hotspot signal handler. Consider using jsig library.");
+      }
+      int this_flag = current_act.sa_flags;
+      LINUX_ONLY(this_flag &= SA_RESTORER_FLAG_MASK;)
+      if (this_flag != expected_act->sa_flags) {
+        st->print_cr("  *** Flags changed! Consider using jsig library.");
```

Comment:
Can't we easily print the difference in flags?

Reply:
Sure. I'll change it.